//defining the message format and syntax
syntax = "proto3";

message personmessage {
  string first_name = 1;
  string last_name = 2;
  string company_name = 3;
  optional int32 id = 4;
}

message PersonListMessage {
  repeated personmessage persons = 1;
}

message PersonRequestMessage {
  optional int32 id = 1;
}
//to retrieve individual and all persons
service PersonService {
  rpc Create(personmessage) returns (personmessage);
  rpc Retrieve(PersonRequestMessage) returns (personmessage);
  rpc RetrieveAll(PersonRequestMessage) returns (PersonListMessage);
}